awx_collection/plugins/modules/group.py

```diff
@@ -172,7 +172,8 @@ def main():
         if (preserve_existing_hosts and relationship == 'hosts') or (preserve_existing_children and relationship == 'children'):
             preserve_existing_check = module.get_all_endpoint(group['related'][relationship])
```

Comment:
On testing where the Inventory exists, and the host exists, but nothing else. This section failed, put in PR to fix this on your branch
https://github.com/darshanip/awx/pull/1j

With both changes, it looks good!